File: GreenSignal/Data/Models/Department.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.Models
{
    public class Department
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [MaxLength(500)]
        public string Name { get; set; }

        [MaxLength(1000)]
        public string AliasNames { get; set; }

        [Required]
        [MaxLength(1000)]
        public string Address { get; set; }

        [Required]
        [MaxLength(100)]
        public string Email { get; set; }

        [Required]
        public bool IsActive { get; set; }

        [Required]
        public Guid LocationId { get; set; }

        [Required]
        public Location Location { get; set; }
    }
}

File: GreenSignal.Mobile/app/src/main/java/com/example/greensignal/domain/model/response/Department.kt
package com.example.greensignal.domain.model.response

import com.example.greensignal.data.remote.dto.response.DepartmentDto

data class Department(
    val id: String = "",
    val name: String = "",
    val aliasNames: String = "",
    val address: String = "",
    val email: String = "",
    val isActive: Boolean = true,
    val locationId: String = "",
    val location: Location = Location()
)

fun DepartmentDto.toDepartment(): Department {
    return Department(
        id = id,
        name = name,
        aliasNames = aliasNames,
        address = address,
        email = email,
        isActive = isActive,
        locationId = locationId,
        location = location.toLocation()
    )
}
